homeassistant/components/hassio/handler.py

```diff
@@ -558,15 +558,22 @@ async def send_command(
 
         This method is a coroutine.
         """
-        url = f"http://{self._ip}{command}"
-        if url != str(URL(url)):
+        url = f"{self._base_url}{command}"
+        joined_url = self._base_url.join(URL(command))
+        normalized_url_str = str(joined_url)
+        # This check is to make sure the normalized URL string
+        # is the same as the URL string that was passed in. If
+        # they are different, then the passed in command URL
+        # contained characters that were removed by the normalization
+        # such as ../../../../etc/passwd
+        if url != normalized_url_str:
```

Comment:
This is already covered by test_send_command_invalid_command

Reply:
This check isn't exactly the same as before, since the base url already is a return value of `URL`.

Reply:
`self._base_url` should always stringily to `http://{self._ip}` so it should be exactly the same. It would be cheaper to keep it as it was before though